python/plugins/processing/algs/qgis/Delaunay.py

```diff
@@ -111,10 +117,14 @@ def processAlgorithm(self, progress):
             attrs = []
             step = 0
             for index in indicies:
-                request = QgsFeatureRequest().setFilterFid(ptDict[ids[index]])
+                fid, n = ptDict[ids[index]]
+                request = QgsFeatureRequest().setFilterFid(fid)
```

Comment:
Thanks Mathias for pointing this. There is also fetching features from this layer also earlier so maybe this could be optimized even more.

Purpose of this pull request wasn't optimizing existing algorithm and I would rather to do this on another one. I also need to test on bigger data to check if this won't change outputs and how it's affecting memory usage.